posts/2022/10-19-graphql_subscription_testing_pt1.md

```diff
@@ -0,0 +1,175 @@
+==title==
+GrapQL subscriptions with Absinthe - testing basic subscriptions
+
+==author==
+Cornelia Kelinske
+
+==description==
+This is part 2 of a 3-part series on setting up and testing GraphQL subscriptions with Absinthe.
+
+==tags==
+coding, elixir, graphql, absinthe, subscriptions
+
+==body==
+
+# 1. Previously on this blog
+
+In [part 1](https://connie.codes/post/graphql_subscription_setup) of this little series, we looked at how we can set up GraphQL subscriptions: the "vanilla" ones that are automatically triggered by a mutation as well as manually triggered subscriptions. In this post, we will write some tests for the former kind. Let's go!
+
+
+# 2. The infrastructure 
+
+First of all, let's have a quick look into our test folder:
+
+```
+├── test
+│   ├── my_app
+│   │   ├── some_other_test.exs
+│   ├── my_app_web
+│   │   ├── some_other_test.exs
+│   │   ├── schema
+│   │   │   ├── mutations
+│   │   │   │   ├── some_mutation_test.exs
+│   │   │   │   └── user_test.exs
+│   │   │   ├── queries
+│   │   │   │   ├── some_query_test.exs
+│   │   │   │   └── user_test.exs
+│   │   │   └── subscriptions
+│   │   │       └── user_test.exs
+│   │   └── views
+│   │       └── error_view_test.exs
+│   ├── support
+│   │   ├── channel_case.ex
+│   │   ├── conn_case.ex
+│   │   ├── data_case.ex
+│   │   ├── fixtures
+│   │   │   └── user_fixtures.ex
+│   │   └── subscription_case.ex
+│   └── test_helper.exs
+```
+
+We see that, as per usual, the structure of our `test` folder mirrors the `lib` folder (shown in the previous post). 
+
+In the `support` folder, we want to make sure that `@endpoint` in `channel_case.ex` is set correctly (it should be by default).
```

Comment:
Are you going to provide an example of it being set correctly?

Reply:
I am now :)